### PR TITLE
mesa 21.2.1

### PR DESCRIPTION
**Pull request by yurikoles**
Created with `brew bump-formula-pr`.

`resource` blocks may require updates.

### CODE REVIEW

File: Formula/mesa.rb [hunk 1]
```diff
@@ -3,14 +3,14 @@ class Mesa < Formula
...
+  url "https://mesa.freedesktop.org/archive/mesa-21.2.1.tar.xz"
+  sha256 "2c65e6710b419b67456a48beefd0be827b32db416772e0e363d5f7d54dc01787"
   license "MIT"
   head "https://gitlab.freedesktop.org/mesa/mesa.git", branch: "main"
 
   livecheck do
-    url "https://www.mesa3d.org/news/"
-    regex(/>\s*Mesa v?(\d+(?:\.\d+)+) is released\s*</i)
+    url "https://mesa.freedesktop.org/archive/"
+    regex(/href=.*?mesa[._-]v?(\d+(?:\.\d+)+)\.t/i)
```
Comment (SMillerDev) [inline Formula/mesa.rb:13]: I'm not sure if this is right, the latter seems like it could be visible without being explicitly released.

Comment (yurikoles) [reply Formula/mesa.rb:13]: Mesa news is always lagging behind mailing list announcement, which is hard to parse, because it contains nested months.

Comment (SMillerDev) [reply Formula/mesa.rb:13]: @samford or @nandahkrishna opinions?

**Comment by cho-m** [comment]
On separate note, can we try changing `libelf` dependency to `elfutils` for Linux?

We only have `mesa` and `dynamips` left to deal with, so it would be nice to see if `mesa` change works here. Otherwise, we would need to rebuild the bottles again (+ revision bump in `linuxbrew-core`).

File: Formula/mesa.rb (hunk 1 above)
Comment (samford) [reply Formula/mesa.rb:13]: Over time, the news page hasn't provided a new post for every released version, so there's something to be said for modifying the `livecheck` block but there are benefits/detriments to each approach:

* Checking the news page ensures we won't update to a new release until it's announced but livecheck will miss any versions that don't get posted (this is an ongoing issue).
* Checking the [directory listing page where the `stable` archive is found](https://archive.mesa3d.org/) will identify all the tarball versions but we may theoretically update to a new version before it's announced.
* It's technically possible to check the [mesa-announce mailing list archive](https://lists.freedesktop.org/archives/mesa-announce/) but it would involve multiple requests (one to fetch the main archive page, another to fetch the page for the newest month). If the page for the newest month doesn't provide versions, we could check the previous month's page but we would have to provide a hard limit to the number of pages we would check before failing.

Checking the mesa-announce mailing list may be ideal from the standpoint of having livecheck report a new version only when it's announced but the implementation would be involved and potentially error-prone.

My view is that we should update this `livecheck` block to check the directory listing where the `stable` archive is found (which this PR currently does) and see how that goes. If we regularly run into a situation where a tarball for a version appears far before the version is announced, then we can always revisit this. However, if it works fine in practice then we would avoid the relative complexity of checking the mesa-announce mailing list archive.

`mesa.freedesktop.org/archive/` redirects to `archive.mesa3d.org`, so I'll add some suggestions to update these URLs to avoid redirecting. Other than that, the regex is good to go.

File: Formula/mesa.rb [hunk 2]
```diff
@@ -3,14 +3,14 @@ class Mesa < Formula
 
   desc "Graphics Library"
   homepage "https://www.mesa3d.org/"
-  url "https://mesa.freedesktop.org/archive/mesa-21.2.0.tar.xz"
-  sha256 "0cb3c802f4b8e7699b1602c08c29d06a4d532ab5b8f7a64676c4ca6bb8f4d426"
+  url "https://mesa.freedesktop.org/archive/mesa-21.2.1.tar.xz"
```
Comment (samford) [inline Formula/mesa.rb:6]:
```suggestion
  url "https://archive.mesa3d.org/mesa-21.2.1.tar.xz"
```

File: Formula/mesa.rb [hunk 3]
```diff
@@ -3,14 +3,14 @@ class Mesa < Formula
...
+  sha256 "2c65e6710b419b67456a48beefd0be827b32db416772e0e363d5f7d54dc01787"
   license "MIT"
   head "https://gitlab.freedesktop.org/mesa/mesa.git", branch: "main"
 
   livecheck do
-    url "https://www.mesa3d.org/news/"
-    regex(/>\s*Mesa v?(\d+(?:\.\d+)+) is released\s*</i)
+    url "https://mesa.freedesktop.org/archive/"
```
Comment (samford) [inline Formula/mesa.rb:12]:
```suggestion
    url "https://archive.mesa3d.org/"
```

**Review by SMillerDev — APPROVED** [review]
